src/common/RawLabel.js

```diff
@@ -32,11 +34,18 @@ export default class RawLabel extends PureComponent<Props> {
   };
 
   render() {
-    const { text, style, ...restProps } = this.props;
+    const { text, children, style, ...restProps } = this.props;
+
+    invariant(
+      (text === undefined && children !== undefined)
+        || (text !== undefined && children === undefined),
```

Comment:
I'd think to write `!!text !== !!children`, but perhaps that's more cryptic.

Reply:
Interesting. It's pretty opaque, to me, and I'm not sure I'd choose it everywhere.

But in this setting, it has the unusual property of not really needing to be understood. `RawLabel`'s callers won't violate the invariant if they've read and understood its jsdoc, and if they do violate it, the error message will tell them clearly what they've done wrong. And there's no other reason the invariant would be violated.

It also has the really nice property of letting us fit the whole `invariant(...)` on a single line. So I'll do it; thanks for the idea! 🙂